api/grpc/mpi/v1/command.proto

```diff
@@ -0,0 +1,358 @@
+    repeated google.protobuf.Struct labels = 3;
+    // A list of features that the NGINX Agent has
+    repeated string features = 4;
+    // Message buffer size, maximum not acknowledged messages from the subscribe perspective
+    string message_buffer_size = 5;
+    // 
+}
+
+// The command settings, associated with messaging from an external source 
+message Command {
+    // The connection and security settingss for the command server
+    ConnectionSettings connection_settings = 1;
+}
+
+// The metrics settings associated with orgins (sources) of the metrics and destinations (exporter)
+message Metrics {
+    // The connection and security settingss for the sources
+    repeated Source sources = 1;
+    // The connection and security settingss for the exporters server
+    repeated Exporter exporters = 2;
+    // the local buffer size that we will cache if connectivity issues exist
+    string bulk_size = 3;
+}
+
+// A source configuration
+message Source {
+    // how often to collect data from a particular source. In google.protobuf.Timestamp format
+    google.protobuf.Timestamp collection_interval = 1;
+    // connection information to connect to a particular source
+    ConnectionSettings connection_settings = 2;
+}
+
+// A destination configuration
+message Exporter {
+    // how often to report in google.protobuf.Timestamp format
+    google.protobuf.Timestamp report_interval = 1;
+    // connection information to send data to a particular destination
+    ConnectionSettings connection_settings = 2;
+}
+
+// A set of connection information and it's associated auth, tls and backoff configurations
+message ConnectionSettings {
+    // Server settings that include connection information
+    Server server = 1;
+    // Authentication settings
+    Auth auth = 2;
+    // Optional TLS settings
+    TLSSetting tls = 3;
+    // backoff settings associated with this connection
+    Backoff backoff = 4;
+}
+
+// Server settings like hostname
+message Server {
+    // the host information
+    string host = 1;
+    // the port information
+    int32 port = 2;
+    // Connection Type enumeration
+    enum ConnectionType {
+        // Default connection type
+        UNKNOWN = 0;
+        // HTTP connection type
+        HTTP = 1;
+        // gRPC connection type
+        GRPC = 2;
+        // unix socket connection type
+        UNIX = 3;
+    }
+    // Specific connection type
+    ConnectionType connection_type = 3;
+}
+
+// Authentication settings
+message Auth {
+    // A token
+    string token = 1;
+}
+
+message TLSSetting {
+    // enable tls
+    bool enable      = 1;
+    // tls cert
+    string cert      = 2;
+    // tls key
+    string key       = 3;
+    // certificate authoirty cert
+    string ca        = 4;
+    // enable verification of a server's certificate chain and host name
+    bool skip_verify = 5;
+}
+
+// backoff settings
+message Backoff {
```

Comment:
this feels like Backoff setting for certain implementation of back-off algorithm/library, recommend simplify to "interval" and "max_elapsed"

Reply:
changed to keepalive